google/resource_storage_bucket_test.go

```diff
@@ -902,6 +929,44 @@ resource "google_storage_bucket" "bucket" {
 `, bucketName)
 }
 
+func testAccStorageBucket_encryption(projectId, projectOrg, projectBillingAccount, keyRingName, cryptoKeyName, bucketName string) string {
+	return fmt.Sprintf(`
+resource "google_project" "acceptance" {
+	name            = "%s"
+	project_id      = "%s"
+	org_id          = "%s"
+	billing_account = "%s"
+}
+
+resource "google_project_services" "acceptance" {
+	project = "${google_project.acceptance.project_id}"
+
+	services = [
+	  "cloudkms.googleapis.com",
+	]
+}
+
+resource "google_kms_key_ring" "key_ring" {
+	project  = "${google_project_services.acceptance.project}"
+	name     = "%s"
+	location = "us"
+}
+
+resource "google_kms_crypto_key" "crypto_key" {
+	name            = "%s"
+	key_ring        = "${google_kms_key_ring.key_ring.id}"
+	rotation_period = "1000000s"
+}
+
+resource "google_storage_bucket" "bucket" {
+	name = "%s"
+	encryption {
+		default_kms_key_name = "projects/${google_project.acceptance.name}/locations/us/keyRings/${google_kms_key_ring.key_ring.name}/cryptoKeys/${google_kms_crypto_key.crypto_key.name}"
```

Comment:
Oh this is sad, but I guess it'll do for now. Maybe in the future we should add a `self_link` or `full_name` or something like that to the kms resources.

Reply:
I thought about doing that, but it doesn't have one in the API... do you think that's still a good idea?  I'm happy to do it - I hate this solution a little bit.

Reply:
Yeah, probably. If we don't, people might realize that the `id` field has the value in the format we want, and I don't want people depending on that. It _is_ in the API under `name`, we've just co-opted the name to mean the name instead of the relative path.

Reply:
FWIW, the KMS team uses these terms:

- `ThingID` - the full resource ID including the projects/...
- `ThingName` - just the things name

e.g.

- `CryptoKeyID` - projects/my-project/locations/global/keyRings/my-keyring/cryptoKeys/my-crypto-key
- `CryptoKeyName` - my-crypto-key

Reply:
Not in the API: https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings
https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys

Reply:
Fixed!